src/mym.cpp

```diff
@@ -1179,6 +1185,23 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
     }
 }
 
+/**
+ * Helper function for checking if string subString is found at the start of string sourceString
+ * @param sourceString Source string to find subString at the start
+ * @param subString String target to look at the beginning of sourceString
+ * @returns Boolean answer if subString was found at the start of sourceString
+ */
+bool isSubstringFountAtTheBeginningCaseInsenstive(const char* sourceString, const char* subString)
+{
+    for (size_t i = 0; subString[i] != '\0'; i++) 
+    {
+        if (tolower(sourceString[i]) != tolower(subString[i]))

```

Comment:
@Synicix There is one potential issue here. There could be whitespace before the first command in the query. This would offset it and it wouldn't match in this logic. We should have a separate index for the `sourceString` to skip until we have no whitespace. Consider this query:
```sql
          CREATE                  TABLE               test          (id          int);
```

Reply:
Wait who would put a white space before the start command? I know that is valid, but that seems like a waste of bytes.

Reply:
It is not necessarily 'who' as opposed to 'what'. There are many utilities nowadays that automatically generate SQL on the fly (DataJoint being one). Therefore, good practice would dictate that we should anticipate any valid SQL and not break if it is 'executable'. Unfortunately, whitespace before a command is valid... I totally agree on the waste of bytes but that is a choice that is not always in our control.

Reply:
Fixed to deal with that behavior.